pkg/cluster/stores.go

```diff
@@ -34,11 +44,23 @@ func NewStoreSet(logger log.Logger, reg *prometheus.Registry, peer *Peer) *Store
 	)
 	reg.MustRegister(met)
 
+	storeNodeConnections := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "thanos_store_nodes_grpc_connections",
+		Help: "Number indicating current number of gRPC connection to store nodes. This indicates also to how many stores query node have access to.",
+	})
+	storeNodeDialHist := prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name: "thanost_store_node_grpc_dial_seconds",
+		Help: "Histogram of block gRPC dialing latency (seconds) until connection is maintained or aborted.",
+		// Possible values: dialSuccess or dialFailure.
+	}, []string{"status"})
+
 	return &StoreSet{
-		logger:      logger,
-		peer:        peer,
-		stores:      map[string]*storeInfo{},
-		grpcMetrics: met,
+		logger:               logger,
+		peer:                 peer,
+		stores:               map[string]*storeInfo{},
+		grpcMetrics:          met,
+		storeNodeConnections: storeNodeConnections,
+		storeNodeDialHist:    storeNodeDialHist,
```

Comment:
Usually the metric type is not part of the name, s/Hist/Duration/ would be a more common name.

Reply:
kk